api/application/src/review/delete.rs

```diff
@@ -1,30 +1,57 @@
 use diesel::prelude::*;
+use domain::{enums::role::Role, models::review::Review};
 use infrastructure::ServerState;
-use rocket::{response::status::NotFound, serde::json::Json, State};
-use shared::response_models::ResponseMessage;
+use rocket::{http::Status, serde::json::Json, State};
+use shared::{response_models::ResponseMessage, token::JWT};
 
 pub fn delete_review(
     review_id: i32,
     state: &State<ServerState>,
-) -> Result<Json<ResponseMessage>, NotFound<Json<ResponseMessage>>> {
+    token: JWT,
+) -> Result<Json<ResponseMessage>, (Status, Json<ResponseMessage>)> {
     use domain::schema::reviews::dsl::*;
+    use domain::schema::reviews;
 
     let pooled = &mut state.db_pool.get().unwrap();
 
+    let review: Review = match pooled.transaction(|c| reviews::table.find(review_id).first::<Review>(c)) {
+        Ok(review) => review,
+        Err(err) => match err {
+            diesel::result::Error::NotFound => {
+                let response = ResponseMessage {
+                    message: Some(String::from("Review could not be found")),
+                };
+
+                return Err((Status::NotFound, Json(response)));
+            }
+            _ => {
+                panic!("Database error - {}", err);
+            }
+        },
+    };
+
+    if token.claims.sub != review.user_id && token.claims.role != Role::Admin {
```

Comment:
Check for admin before accessing DB.

Reply:
This happens as we need the user id tied to the review before we can verify who they are